### ISSUE TITLE
11.1.4.3 / 11.1.4.11 Kontrastmodus des Betriebssystems akzeptieren

### ISSUE THREAD

**Issue by johannesFischer84** [BIK-BITV/BIK-App-Test#87]
Ich möchte anregen, dass in den Kontrastprüfschritten für Apps (WCAG 1.4.3 und 1.4.11) zur Konformität auch die Einstellung des Kontrastmodus (oder Hell-/Dunkel-Modus) akzeptiert wird. In der Regel akzeptieren wir Konformität mit entsprechenden Einstellungen des Betriebssystems, z. B. in EN 11.7, bei der Zoom-Vergrößerung oder vielleicht auch bei der Pausierung von Animationen. Zudem haben besonders unter iOS native Bedienelemente in der Standard-Einstellung zu geringen Kontrast, was sich in der hohe-Kontrast-Einstellung aber anders darstellen kann.

Wenn wir es akzeptieren, müsste es in die Prüfanleitungen aufgenommen werden.

**Comment by detlevhfischer** [comment]
@johannesFischer84 Hieße das für Dich, grundsätzlich mit dieser Vorauswahl zu testen? 
Ein Argument dafür ist, dass sehbehinderte Menschen dies ja grundsätzlich einstellen können.
Das Gegenargument ist, dass es viel Nutzer (?) ggf nicht wissen ,gerade die Älteren, die zunehmend Probleme bekommen, aber teczhnisch nicht versiert sind.
Mit der Grundeinstellung wäre dafür gesporgt, dass native iOS Switches dann automatisch ausreichen würden. Anderenfalls zu verlangen, dass Entwickler nicht native Bedienelemente nutzen sollen, da diese im Default-Zustand die Kontrastanforderung nicht erfüllen, käme mir fasch vor.

**Comment by johannesFischer84** [comment]
@detlevhfischer Detlev, ich stimme dir zu. Ich würde schon zunächst mit dem Standardmodus testen. Aber wenn man auf Fehler stößt, kann man den Hohen-Kontrast-Modus bzw. in Kombination den Wechsel von Hell-Modus und Dunkel-Modus ausprobieren. Wenn es Fehler im Standard-Modus gibt, die über die Einstellungen mit anderen Modus behoben werden können, würde ich den Betreiber/Entwickler auch immer darauf hinweisen, dass es noch nicht perfekt ist, eben weil nicht alle Nutzenden über die Einstellungen Bescheid wissen (Info wäre über eine App-Hilfe-Funktion möglich). Es könnte aber noch als konform akzeptiert werden.
Ich denke auch, dass Entwickler soweit irgendwie möglich auf native Standard-Elemente der Umgebung setzen können sollten.

**Comment by mitchellevan** [comment]
Es gibt seit langem Präzedenzfälle, dass der Kontrast nur mit Standardeinstellungen getestet werden sollte. Aus eigener Erfahrung weiß ich, dass Level Access und TPGi auf diese Weise testen. Weitere Beispiele:

* [Testing Color Contrast in Mobile Apps (Deque Blog)](https://www.deque.com/blog/testing-color-contrast-in-mobile-apps/): "Even though it’s great to support these settings in your apps, you cannot use them as a way to meet WCAG criteria."
* Beim Testen von 1.4.3 und 1.4.11 ändert die [Testmethodik von IBM](https://www.ibm.com/able/requirements/requirements/) die Systemeinstellungen nicht.

Wenn sich diese Präzedenzfälle geirrt hätten, würde ich gerne bessere vorschlagen. Die Präzedenzfälle sind jedoch gut für die Benutzer und nicht unlogisch.

> ...dass es viel Nutzer (?) ggf nicht wissen ,gerade die Älteren, die zunehmend Probleme bekommen, aber teczhnisch nicht versiert sind.

**Comment by detlevhfischer** [comment]
Ich habe die Ausnahme für native controls wie iOS switches vorerst rausgenommen und ein Issue bei dee WCAG2ICT Task Force angelegt: https://github.com/w3c/wcag2ict/issues/94

**Comment by detlevhfischer** [comment]
Es gab dazu schon ein älteres Issue https://github.com/w3c/wcag2ict/issues/82

Ich habe daraufhin den [Prüfschritt zu 11.1.4.11](https://github.com/BIK-BITV/BIK-App-Test/blob/master/Pr%C3%BCfschritte/de/11.1.4.11%20Kontraste%20von%20Grafiken%20und%20Bedienelementen%20ausreichend.adoc) vorläufig angepasst und nochmal in einem [Kommentar ](https://github.com/w3c/wcag2ict/issues/82#issuecomment-1432822831)um Bestätigung dieser Position durch die WCAG2ICT task force gebeten.

**Comment by detlevhfischer** [comment]
Kann ich dieses Issue schließen? Der Prüfschritt hat jetzt eine Ausnahme für unveränderte Bedienelemente wie iOS Switches (die Nutzende, wenn sie die Einstellung kennen, über "Kontraste erhöhen" konform darstellen können). 

@mitchellevan Damit weichen wir ggf. von den von Dir genannten Verfahren von Deque und IBM ab. 
IBM hat allerdings bei 1.4.11:

> "UI components whose appearance is entirely determined by the user agent (e.g., stock HTML radio buttons and checkboxes) do not need to meet the 3:1 contrast if they are not modified by authors." 

Wenn man für _user agent_ hier _OS_ einsetzt, würde das auch auf native iOS switches zutreffen, denke ich.

**Comment by johannesFischer84** [comment]
@detlevhfischer 
> Kann ich dieses Issue schließen? Der Prüfschritt hat jetzt eine Ausnahme für unveränderte Bedienelemente wie iOS Switches (die Nutzende, wenn sie die Einstellung kennen, über "Kontraste erhöhen" konform darstellen können).

Bei 11.1.4.11 habe ich die Ausnahme gefunden. Danke. Wenn zusätzlich zum Symbol auf dem Schalter aber auch Text enthalten ist, sollte die Ausnahme auch bei 11.1.4.3 eingefügt werden, finde ich. Was meinst du?

OffTopic: Entschuldige bitte den Zeitverzug. Ich habe irgendwie übersehen, dass du nach dem Schließen gefragt hast und ich ja das Issue eröffnet hatte. Ich habe dieses Issue gerade nur zufällig in anderem Zusammenhang geöffnet. Aber dann habe ich jetzt gleich mal geantwortet.

**Comment by mitchellevan** [comment]
> @mitchellevan Damit weichen wir ggf. von den von Dir genannten Verfahren von Deque und IBM ab.

Die Testschritte in 'main' sind nun korrekt, da sie nur unter 1.4.11 eine Ausnahme für unveränderte native Bedienelemente vorsehen, nicht aber unter 1.4.3. Ich sehe keine Abweichung von den Testverfahren der anderen Firmen.

**Comment by detlevhfischer** [comment]
@mitchellevan Ich denke das Issue kann geschlossen werden. Sonst gerne wieder öffnen.
